tests/src/test/scala/whisk/core/containerpool/logging/ElasticSearchLogStoreTests.scala

```diff
@@ -0,0 +1,237 @@
+  private val defaultHttpRequest = HttpRequest(method = GET, uri = "https://some.url", entity = HttpEntity.Empty)
+  private val defaultPayload = JsObject(
+    "query" -> JsObject(
+      "query_string" -> JsObject("query" -> JsString(
+        s"_type: ${defaultConfig.logMessageField} AND ${defaultConfig.activationIdField}: $activationId"))),
+    "sort" -> JsArray(JsObject("time_date" -> JsObject("order" -> JsString("asc"))))).compactPrint
+  private val defaultUri = Uri(s"/whisk_user_logs/_search")
+
+  private var expectedHeaders: List[HttpHeader] = defaultHeaders
+  private var expectedHttpResponse = defaultHttpResponse
+  private val expectedLogs = ActivationLogs(
+    Vector("2018-03-05T02:10:38.196689522Z stdout: some log stuff", "2018-03-05T02:10:38.196754258Z stdout: more logs"))
+  private var expectedPayload = defaultPayload
+  private var expectedUri = defaultUri
+
+  private val activation = WhiskActivation(
+    namespace = EntityPath("namespace"),
+    name = EntityName("name"),
+    Subject(),
+    activationId = activationId,
+    start = ZonedDateTime.now.toInstant,
+    end = ZonedDateTime.now.toInstant,
+    response = ActivationResponse.success(Some(JsObject("res" -> JsNumber(1)))),
+    logs = expectedLogs,
+    annotations = Parameters("limits", ActionLimits(TimeLimit(1.second), MemoryLimit(128.MB), LogLimit(1.MB)).toJson))
+
+  private val testFlow
+    : Flow[(HttpRequest, Promise[HttpResponse]), (Try[HttpResponse], Promise[HttpResponse]), NotUsed] =
+    Flow[(HttpRequest, Promise[HttpResponse])]
+      .mapAsyncUnordered(1) {
+        case (request, userContext) =>
+          request.headers shouldBe expectedHeaders
+          request.uri shouldBe expectedUri
+
+          Unmarshal(request.entity)
+            .to[String]
+            .map { payload =>
+              payload shouldBe expectedPayload
+
+              (Success(expectedHttpResponse), userContext)
+            }
+            .recover {
+              case e =>
+                (Failure(e), userContext)
+            }
+      }
+
+  override def beforeEach = {
+    expectedHeaders = defaultHeaders
+    expectedHttpResponse = defaultHttpResponse
+    expectedPayload = defaultPayload
+    expectedUri = defaultUri
+  }
```

Comment:
Have you checked my comments in #3416 towards possible alternatives here? (and the relevant tests)